### PR TITLE
chore: new filter tests

### PR DESCRIPTION
**Pull request by fbarbu15**
## Problem
After reviewing the filter tests found the following problems:
- Only had 6 automated tests
- Asserts inside callbacks were not working (not failing tests)
- BeforeEach code was sometimes failing when starting  docker containers
- There was no reusable functions in regards with checking messages and doing asserts

## Solution

Added PR with following fixes:
- Reached 65 automated tests
- Moved asserts out of the callbacks so that they work properly now
- Added retries on failing code from before each
- Added some reusable functions regarding message collection and validation

## Notes

- There's a test that randomly fails and left it skipped for now. Not sure if I implemented it badly or it's an actual issue (known or not): `Subscribe and receive messages from multiple nwaku nodes` 
- There are 2 failing tests that are skipped and related to: https://github.com/waku-org/js-waku/issues/1464

### CODE REVIEW

File: packages/tests/tests/filter.node.spec.ts [hunk 1]
```diff
@@ -11,203 +13,961 @@ import { createLightNode } from "@waku/sdk";
...
+      console.log("jswaku node failed to start:", error);
+    }
     await waku.dial(await nwaku.getMultiaddrWithId());
     await waitForRemotePeer(waku, [Protocols.Filter, Protocols.LightPush]);
     subscription = await waku.filter.createSubscription();
+    messageCollector = new MessageCollector();
+  });
+
+  it("Subscribe and receive messages via lightPush", async function () {
+    // Subscribe to the content topic and set callback for received messages.
```
Comment (weboko) [inline packages/tests/tests/filter.node.spec.ts:253]: It's good to have comments but to me it seems we can drop some of them sicne it is clear what's happening by the code.

File: packages/tests/tests/filter.node.spec.ts [hunk 2]
```diff
@@ -2,7 +2,9 @@ import {
   createDecoder,
```
Comment (weboko) [inline packages/tests/tests/filter.node.spec.ts:2]: Amazing job! I have only one comment here - the file is huge now. 
I propose to break it into some files under `filter` folder.

File: packages/tests/tests/filter.node.spec.ts (hunk 1 above)
Comment (fbarbu15) [reply packages/tests/tests/filter.node.spec.ts:253]: Agreed, will remove some of them :)

File: packages/tests/tests/filter.node.spec.ts (hunk 2 above)
Comment (fbarbu15) [reply packages/tests/tests/filter.node.spec.ts:2]: Yes, makes sense, I'll create a folder with multiple files based on features:
- subscribe
- ping
- unsubscribe
- filterPush
- filterTestUtils
Makes sense?

**Review by weboko — APPROVED** [review]

**Pull request merged**